### PR TITLE
Upgrade ag-grid to 31.x

### PR DESCRIPTION
**Pull request by cnrudd**
This can be reviewed in Toolbox with its corresponding PR: https://github.com/xh/toolbox/pull/709
This is a breaking change even if an app made no use of `ColumnApi` (which has been removed) because the version getting logic in an app's`Bootstrap.ts` file will need to be redone as in the [Toolbox example.](https://github.com/xh/toolbox/pull/709/files/5626e21d778e1fc72f9735d2d8f011513e1ac9c6#diff-304055320a29f66ea1255446ba8f13e0f3f1b13643bcea0c0466aa60e9288a8f)

See [What's New in AG Grid 31](https://blog.ag-grid.com/whats-new-in-ag-grid-31/) and [Upgrading to AG Grid 31](https://www.ag-grid.com/javascript-data-grid/upgrading-to-ag-grid-31/?ref=blog.ag-grid.com) for more details.

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

**Comment by lbwexler** [comment]
All set to merge this -- @cnrudd -- you mentioned an issue you wanted to discuss with atm?

File: cmp/ag-grid/AgGridModel.ts [hunk 1]
```diff
@@ -574,7 +573,7 @@ export class AgGridModel extends HoistModel {
      */
     setPinnedTopRowData(data: PlainObject[]) {
         this.throwIfNotReady();
-        this.agApi.setPinnedTopRowData(data);
+        this.agApi.setGridOption('pinnedTopRowData', data);
     }
```
Comment (lbwexler) [inline cmp/ag-grid/AgGridModel.ts:577]: @cnrudd -- think we missed the matching change for `setPinnedBottomRowData`

**Comment by lbwexler** [comment]
@cnrudd  -- what would you think about using the applyGridOptions({}) form even if we are setting just a single option?  I think it would be more readable and consistent across our code and we would not have to specify the property name as a string, which seems so clunky to me.

File: cmp/ag-grid/AgGridModel.ts (hunk 1 above)
Comment (cnrudd) [reply cmp/ag-grid/AgGridModel.ts:577]: Fixed. thanks!

**Comment by cnrudd** [comment]
> @cnrudd -- what would you think about using the applyGridOptions({}) form even if we are setting just a single option? I think it would be more readable and consistent across our code and we would not have to specify the property name as a string, which seems so clunky to me.

**Review by lbwexler — COMMENTED** [review]
I am all set to merge these

**Pull request merged**